benchmarks/README.md

```diff
@@ -0,0 +1,104 @@
+scavenged from `/benchmarks`, which injects Envoy between the 
+benchmark client and test server.
+
+```bash
+git clone https://github.com/oschaaf/nighthawk.git benchmark-test
+cd benchmark-test
+bazel build //benchmarks:benchmarks
+
+# Specify the ip address family we'll be using. [v4only|v6only|all]
+export ENVOY_IP_TEST_VERSIONS=v4only
+# Explicit tmpdir for OSX docker, to make sure we'll use a volume that works when
+export TMPDIR="$(pwd)/benchmarks/tmp"
+# Nighthawk tools will be sourced from this docker image
+export NH_DOCKER_IMAGE="envoyproxy/nighthawk-dev:latest"
+# Envoy docker image that we'll use to inject the Envoy proxy
+export ENVOY_DOCKER_IMAGE_TO_TEST="envoyproxy/envoy-dev:74290ef76a76fbbf50f072dc33438791f93f68c7"
+# Envoy is called 'Envoy' in the Envoy docker image.
+export ENVOY_PATH="envoy"
+
+# run all tests starting with test_http_h1_small in benchmarks/
+bazel-bin/benchmarks/benchmarks --log-cli-level=info -vvvv -k test_http_h1_small benchmarks/
+```
+
+## Example: running with binaries
+
+This will build the Nighthawk binaries from the c++ code, and use those to
+execute the benchmarks. Environment variable `ENVOY_PATH` can be used to
+specify a custom Envoy binary to use to inject as a proxy between the test
+client and server. If not set, the benchmark suite will fall back to configuring
+Nighthawk's test server for that. Note that the build can be a lengthy process.
+
+```bash
+git clone https://github.com/oschaaf/nighthawk.git benchmark-test
+cd benchmark-test
+bazel test --test_summary=detailed --test_output=all --test_arg=--log-cli-level=info --test_env=ENVOY_IP_TEST_VERSIONS=v4only --test_env=HEAPPROFILE= --test_env=HEAPCHECK= --cache_test_results=no --compilation_mode=opt --cxxopt=-g --cxxopt=-ggdb3 //benchmarks:*
+```
+
+## Example: fully dockerized flow
+
+The framework can be run via docker and used that way to execute
+python benchmarks scripts not sourced from this repository, but
+elsewhere. An example:
+
+```bash
+# This script runs the dockerized benchmarking framework, which in
+# turn will pull Nighthawk and Envoy in via docker. 
+
+set -eo pipefail
+set +x
+set -u
+
+# The benchmark logs and artifacts will be dropped here
+OUTDIR="/home/oschaaf/code/envoy-perf-vscode/nighthawk/benchmarks/tmp/"
+# Used to map the test that we want to see executed into the docker container
+# Note: the contents could be fetched via http, for example.
+TEST_DIR="/home/oschaaf/code/envoy-perf-vscode/nighthawk/benchmarks/test/"
+
+# Rebuild the docker in case something changed.
+./docker_build.sh && 
+docker run -it --rm \
+  -v "/var/run/docker.sock:/var/run/docker.sock:rw" \
+  -v "${OUTDIR}:${OUTDIR}:rw" \
+  -v "${TEST_DIR}:/usr/local/bin/benchmarks/benchmarks.runfiles/nighthawk/benchmarks/external_tests/" \
+  --network=host \
+  --env NH_DOCKER_IMAGE="envoyproxy/nighthawk-dev:latest" \
+  --env ENVOY_DOCKER_IMAGE_TO_TEST="envoyproxy/envoy-dev:f61b096f6a2dd3a9c74b9a9369a6ea398dbe1f0f" \
```

Comment:
How do you specify a custom Envoy binary in the Docker flow?

Reply:
This is documented in the flow with binaries. You don't specify a docker image, but instead just use `ENVOY_PATH` to point out your Envoy binary.

Reply:
That certainly is possible, but it result in a large image (I assume the binaries would need to be unstripped and build with all the right flags for profiling?) .

Reply:
Yeah. I don't think it needs to be in this PR, but it's a pretty useful feature IMHO. One thing that I wanted to do is run the exact same Docker image on my local workstation and different sized Cloud VMs to see what performance range is observed.

Reply:
Added this to the list of things that need to be done in `README.md`